src/pybamm/solvers/idaklu_solver.py

```diff
@@ -736,22 +742,30 @@ def _integrate(self, model, t_eval, inputs_dict=None, t_interp=None):
             The model whose solution to calculate.
         t_eval : numeric type
             The times at which to stop the integration due to a discontinuity in time.
-        inputs_dict : dict, optional
+        inputs_list: list of dict, optional
             Any input parameters to pass to the model when solving.
         t_interp : None, list or ndarray, optional
             The times (in seconds) at which to interpolate the solution. Defaults to `None`,
             which returns the adaptive time-stepping times.
         """
-        inputs_dict = inputs_dict or {}
-        # stack inputs
-        if inputs_dict:
-            arrays_to_stack = [np.array(x).reshape(-1, 1) for x in inputs_dict.values()]
-            inputs = np.vstack(arrays_to_stack)
+        inputs_list = inputs_list or [{}]
+
+        # stack inputs so that they are a 2D array of shape (number_of_inputs, number_of_parameters)
+        if inputs_list and inputs_list[0]:
+            inputs = np.vstack(
+                [
+                    np.hstack([np.array(x).reshape(-1) for x in inputs_dict.values()])
+                    for inputs_dict in inputs_list
+                ]
+            )
         else:
             inputs = np.array([[]])
 
-        y0full = model.y0full
-        ydot0full = model.ydot0full
+        # stack y0full and ydot0full so they are a 2D array of shape (number_of_inputs, number_of_states + number_of_parameters * number_of_states)
+        # note that y0full and ydot0full are currently 1D arrays (i.e. independent of inputs), but in the future we will support
+        # different initial conditions for different inputs (see https://github.com/pybamm-team/PyBaMM/pull/4260). For now we just repeat the same initial conditions for each input
+        y0full = np.vstack([model.y0full] * len(inputs_list))
+        ydot0full = np.vstack([model.ydot0full] * len(inputs_list))
```

Comment:
The `y0full` values include the initial values of the differential terms, which may depend on input parameters. These should all be initialized

Reply:
these should have already been initialised during the setup correct?

Reply:
By setup, do you mean in `IDACalcIC`? This gets handled during the call to `_set_consistent_initialization`, but initial values of differential terms cannot be resolved by `IDACalcIC`

Reply:
no, I did mean _set_consistent_initialization. But this function only has a single input dict, if you pass in a list of parameters that affect the initial condition you get an error. Part of #4260 was that if you pass an input list (that affects the initial conditions) this would be handled appropriately so that y0full is a 2d array where the rows are possible different corresponding to different parameters in the list (unlike now where each row is identical)

Reply:
Ah ok, got it